src/contentmanagerside.cpp

```diff
@@ -92,7 +93,14 @@ void ContentManagerSide::setContentManager(ContentManager *contentManager)
 
 void ContentManagerSide::setCategories(QStringList categories)
 {
-    mp_categories->setSelections(categories, KiwixApp::instance()->getSettingsManager()->getCategoryList());
+    const auto lang = QLocale().name().split('_')[0].toStdString();
+    KiwixChoiceBox::SelectionList sList;
+    for (const auto& category : categories) {
+        const auto translation = QString::fromStdString(kiwix::translateBookCategory(lang, category.toStdString()));
+        sList.append({translation, category});
+    }
```

Comment:
I can see the following opportunity for further simplification:

1. Change the parameter type of `SettingsManager::setCategory()` and the return type of `SettingsManager::getCategoryList()` from `FilterList` to `QStringList` (representing only the category identifiers; the content of `SettingsManager::m_categoryList` will change respectively)
2. Convert this piece of code to a helper function in an unnamed namespace `SelectionList addCategoryTranslations(QStringList categoryList)`
3. Then use this function to translate both the full list of categories and the active/default selections for the call of `mp_categories->setSelections(...)` on the next line (and undo the translation in `SettingsManager::initCategoryList()`)

Reply:
Done